### DISCUSSION TITLE
Sidebar required

### DISCUSSION THREAD

**Discussion by ShaunHalliday** [twilio-labs/paste#2209] [Ideas] [answered]
Hi All,

We are currently developing systems for clients using Paste (and other Twilio services) and to keep to the Paste standard we require a sidebar component. I cannot currently find anything in Paste to match this requirement, although an example of it is on Paste site: -

![TwilioSidebar](https://user-images.githubusercontent.com/85227673/153851849-e8953ea2-443c-4294-93cd-ab4380acb857.png)

Also, sizeSidebar is applicable to width properties (example in below screenshot) in a Box component so I was wondering if Sidebar does exist but is not yet released: -

![SidebarWidthSetting](https://user-images.githubusercontent.com/85227673/153852214-c3270032-a7bc-478c-8600-0f75a03b9e77.png)

Thank you once again,

Shaun

**Comment by TheSisb** [comment, marked as answer]
Hello @ShaunHalliday,

That's very exciting, we'd love to see what you build if it can be shared publicly!

We do not currently have a Sidebar component on our [Roadmap](https://paste.twilio.design/roadmap), however you can [see how we put it together by composing Paste components](
https://github.com/twilio-labs/paste/tree/main/packages/paste-website/src/components/site-wrapper/sidebar) for our website. 

> Also, sizeSidebar is applicable to width properties (example in below screenshot) in a Box component so I was wondering if Sidebar does exist but is not yet released: -

Good catch! While we do have the `sizeSidebar` token, it isn't foreshadowing a component release. We added that token to improve consistency in how UIs size their sidebars, so different products maintain a similar look in their layouts. We actually don't have any strong opinions about the rest of how a Sidebar is built, which is why we haven't shipped the encapsulating component. As a system, we tend to focus on [smaller components that can be easily composed](https://paste.twilio.design/introduction/about-paste#paste-is) to build things like Sidebars. 

Hope this helps clarify how we think about components and composition!
